modules/effects/src/main/scala/dev/profunktor/redis4cats/effects.scala

```diff
@@ -40,6 +40,44 @@ object effects {
   final case class ZRange[V](start: V, end: V)
   final case class RangeLimit(offset: Long, count: Long)
 
+  sealed trait ScriptOutputType {
+    type Return[V]
+    private[redis4cats] type Underlying[V]
+    private[redis4cats] val outputType: JScriptOutputType
+    private[redis4cats] def convert[V](in: Underlying[V]): Return[V]
+  }
+  object ScriptOutputType {
+    import dev.profunktor.redis4cats.JavaConversions._
+
+    case object Integer extends ScriptOutputType {
+      type Return[V]                         = Long
```

Comment:
All this ceremony here is needed to make both Lettuce and the Scala compiler happy, check the types in the `eval` implementation.